### DISCUSSION TITLE
Remove completed torrents from qBittorrent.

### DISCUSSION THREAD

**Discussion by fexofenadine** [pymedusa/Medusa#8562] [Ideas & Feature requests]
Feature request [1139](https://github.com/pymedusa/Medusa/issues/1139) suggested that qBitTorrent does not have the API required to remove torrents on completion, however Radarr and Sonar can do it ok, has the API been added now? Is it possible to revisit the feature request now?

**Describe the solution you'd like**
Please add completed download handling for qBitTorrent

**Describe alternatives you've considered**
Using another client, namely Transmission or Deluge, but these clients have drawbacks of their own

**Comment by BenjV** [comment]
Qbittorent does have an API
See:

https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)

**Comment by Rouzax** [comment]
Why not use the qBittorrent option for that?
![image](https://user-images.githubusercontent.com/4103090/95601110-53d58e80-0a53-11eb-9205-593fc1531834.png)

**Comment by fexofenadine** [comment]
Because I don't want to remove all my torrents!

**Comment by bugsyb** [comment]
qBittorrent has the API call to remove torrent.
The way Sonarr/Radarr works is;
-  once seeding finishes according to general settings or per torrent settings (it checks if status is complete, as has the UUID of the torrent - but couldn't quickly find it in qBittorrent API list - for sure Sonarr sources will nicely reveal this little quirk of logic)
- it hardlinks/moves (or other preferred action as per settings could be on Medusa side) to destined location (/tv/?). Hardlink is most suitable as seeding is completed and makes it easier for next step
- removes the torrent.

https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)#remove-item
